src/gus-markdown-extension.js

```diff
@@ -48,14 +60,37 @@ var userStoryDetailClassicID = 'userStoryDetailPage_userStoryWorkForm_detailsInp
 var bugDetailClassicLocation = '/apex/adm_bugdetail';
 var bugDetailClassicID = 'bugDetailPage:bugWorkForm:j_id89bugDetailPage:bugWorkForm:j_id89_00NB0000000FiIs_div';
 
-var lightningBugEdit;
+var investigationDetailClassicLocation = '/apex/adm_investigationdetail';
+var investigationDetailClassicID = 'investigationDetailPage:investigationWorkForm:j_id88investigationDetailPage:investigationWorkForm:j_id88_00NB0000000FiIs_div';
+
+var lightningBugOrInvestigationEdit;
 var iframeContext = false;
 window.gusMarkdownRun = true;
 var originalHTML = '';
 var originalHTMLFromBackground = {};
+var validLocation = false;
+var validLocationRegex = /sObject\/\w{18}\/view|one.app#\w{488}/g;
+
+// First check we run before any attempt to run the extension logic. Makes sure we are on a page that the extension would run on.
+function checkValidLocation () {
+    // If the url contains alohaRedirect, the page is basically in the process of loading, so we tell the background page. The
+    // background page will continually respond to this message, telling this script to run initialize again. Eventually, the url
+    // won't contain alohaRedirect and we will know if the page is valid or invalid
+    if (location.href.indexOf('alohaRedirect') > -1) {
```

Comment:
throw in a set time out so we aren't hammering it constantly, maybe 500 or 1000?